src/app/InteractionModelEngine.cpp

```diff
@@ -1592,9 +1592,31 @@ void InteractionModelEngine::RemoveDuplicateConcreteAttributePath(SingleLinkedLi
     while (path1 != nullptr)
     {
         bool duplicate = false;
+#if CHIP_CONFIG_USE_DATA_MODEL_INTERFACE && CHIP_CONFIG_USE_EMBER_DATA_MODEL
```

Comment:
Could these blocks of code with ifdefs be moved to anonymous namespace methods so that the ifdefing is not inlined?

Reply:
Updated to a method that is more self-contained.